test/corpus.py

```diff
@@ -0,0 +1,45 @@
+import tempfile
+import shutil
+from os import path
+from subprocess import call
+import atexit
+import notmuch
+
+
+class Corpus:
+    def __init__(self):
+        self.TESTDIR = tempfile.mkdtemp()
+        self.TESTCORPUS = path.join(self.TESTDIR, 'corpus')
+        HERE = path.dirname(path.realpath(__file__))
+
+        # copy over corpus
+        shutil.copytree(path.join(HERE,'corpus'), self.TESTCORPUS)
+
+        # create a temporary notmuch database
+        # create notmuch config file
```

Comment:
That is a dependency on the filesystem and on notmuch. That means, this test is not an unit test, rather an functional test (i think). An Unit Test should, according to the book [The Art of Unit Testing](http://artofunittesting.com/), have no real dependencies on anything. The reason is that it dependends on the correct working of the filesystem and notmuch - so if the unit test fails, we don't know whether this bug is really in OUR codebase.